### ISSUE TITLE
[Arm64] Assert in crossgen of Microsoft.CodeAnalysis.dll

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#12199]
This is a new crossgen failure in `8763ef5f`
```
./crossgen /MissingDependenciesOK /Platform_Assemblies_Paths $PWD $PWD/Microsoft.CodeAnalysis.dll

Assert failure(PID 41690 [0x0000a2da], Thread: 41690 [0xa2da]): 
    Assertion failed 'source->isContained()' 
        in 'Microsoft.Cci.PeWriter:WritePeToStream(ref,ref,ref,ref):bool:this' (IL size 908)

    File: src/jit/codegenarmarch.cpp Line: 558
```

**Comment by sdmaclea** [comment]
Last know good version `0d7c2d1`

@dotnet/arm64-contrib @dotnet/arm32-contrib FYI

@CarolEidt Is this likely related to "Make containedness explicit"

**Comment by CarolEidt** [comment]
> Is this likely related to "Make containedness explicit"

Yes, quite possibly. This is crossgen of arm64, right? I'll have a look.

**Comment by sdmaclea** [comment]
Yes it is crossgen of arm64.

**Comment by sdmaclea** [comment]
Fixed by #12189 

@CarolEidt @mskvortsov Thanks
